### DISCUSSION TITLE
V3 -> V4 Migration | LavalinkNodeOptions

### DISCUSSION THREAD

**Discussion by Wellendox** [lavalink4net/Lavalink4NET#142] [Q&A] [answered]
Reading through [the migration guide](https://lavalink4net.angelobreuer.de/docs/migration/V4%20Guide) I stumble upon a single question.

How in the world are we supposed to authenticate the bot with lavalink now that there are no options ANYWHERE to tell the bot which lavalink hostname url, websocket url or password to use?

In V3, I had the LavalinkNodeOptions which I could apply. Now where would I do that now? Seems like a HUGE oversight. I can't be the only one looking for a way to add the options anywhere.

**Comment by angelobreuer** [comment, marked as answer]
Hi, @Wellendox ,

you can configure the node options now using the service collection like the following:

```cs
services.ConfigureLavalink(x =>
{
    x.Label = "Lavalink";
    x.Passphrase = "password"; 
    x.ResumptionOptions = new LavalinkSessionResumptionOptions(TimeSpan.FromSeconds(15));
    x.ReadyTimeout = TimeSpan.FromSeconds(15);
})
```

I will add it to the migration guide later, thank you

**Reply by Wellendox** [reply]
Perfect! I was looking for that extension method.
